### DISCUSSION TITLE
Error with a snippet

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#224] [General]
Hi Franky,
I used a snippet for several month with success. But for 2 weeks I got an error. The snippet add some informations on the user profil page. It use the function eme_get_members_by_wpid_membershipid() but it is now undefined ? Could you help me ?
Here is the snippet 
```add_action('edit_user_profile', 'my_eme_user_profile') ;
add_action('show_user_profile', 'my_eme_user_profile') ;
function my_eme_user_profile($user){
	$membership_id=2;
	$wp_id=get_current_user_id();
	//$person_id = eme_get_personid_by_wpid($user->ID);
	//$memberships=eme_get_activemembership_names_by_personid($person_id);
	$member = eme_get_members_by_wpid_membershipid($wp_id, $membership_id);
	if (!empty($member)) {
		$member=$member[0];
		
		$answers=eme_get_member_answers($member['member_id']);
		
		$coupon=$answers[0];
		$conso=$answers[1];
	?>
	<h3>Carte de réduction</h3>
	<li>Nombre de coupons disponibles : <?php echo $coupon['answer'] ?></li>
	<li> Dates des navigations payées par carte : <?php echo $conso['answer'] ?></li>
	<?php
	}
}```

**Comment by liedekef** [comment]
The function call is (in your case):
`$member = eme_get_member_by_wpid_membershipid( $wp_id, $membership_id, EME_MEMBER_STATUS_ACTIVE . ',' . EME_MEMBER_STATUS_GRACE );`
(_member_, not _members_) and it returns the member directly (not an array of members, so 
`		$member=$member[0];`
is not needed. The function eme_get_members_by_wpid_membershipid was never used in my code so it got removed ...

The next version will no longer require you to specify the status as well, see https://github.com/liedekef/events-made-easy/commit/917cc65cb0ba445b96d440d30c41e8ae41f4b5d6

**Reply by f14m07** [reply]
My snippet certainly worked before. I've taken your advice into account and modified my snippet accordingly. It's perfect, thank you for your support

**Reply by liedekef** [reply]
Yes, your snippet will have worked before, but like I said: the function eme_get_members_by_wpid_membershipid was never used in my code so it got removed (about 3 weeks ago). The doc on the EME site has already been updated to comply with the next version on this.
